packages/jellyfish-api-core/src/category/loan.ts

```diff
@@ -131,6 +140,17 @@ export interface SetCollateralToken {
   activateAfterBlock?: number
 }
 
+export interface CollateralTokensData {
+  [key: string]: CollateralTokenDetail
+}
+
+export interface CollateralTokenDetail {
+  token: string
+  factor: BigNumber
+  priceFeedId: string
+  activateAfterBlock: BigNumber
```

Comment:
optional. https://github.com/DeFiCh/ain/blob/a43fda7d0842ff1c90ab39880df38077ba36bd91/src/masternodes/rpc_loan.cpp#L15

Reply:
For setCollateralToken, yes, activateAfterBlock is optional.  However, listCollateralTokens always return activateAfterBlock value. If you don't pass in activateAfterBlock in the setCollateralToken, you will retrieve the current block when the setCollateralToken executed.

Reply:
yeah true. https://github.com/DeFiCh/ain/blob/13cc9108141fdf4ed0bf93adf8b4b61fd516ebd0/src/masternodes/mn_checks.cpp#L1830
ignore.

Reply:
Anyway I improve the test I wrote earlier